plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -91,6 +97,151 @@ export class GitLabClient {
     return this.pagedRequest(`/groups`, options);
   }
 
+  async listSaasUsers(groupPath: string): Promise<PagedResponse<GitLabUser>> {
+    const items: GitLabUser[] = [];
+    let hasNextPage: boolean = false;
+    let endCursor: string | null = null;
+
+    do {
+      const response: GitLabSaasUsersResponse = await fetch(
+        `${this.config.baseUrl}/api/graphql`,
+        {
+          method: 'POST',
+          headers: {
+            ...getGitLabRequestOptions(this.config).headers,
+            ['Content-Type']: 'application/json',
+          },
+          body: JSON.stringify({
+            variables: { group: groupPath, endCursor },
+            query: `query($group: ID!, $endCursor: String) {
+              group(fullPath: $group) {
+                groupMembers(first: 100, relations: [DESCENDANTS], after: $endCursor) {
+                  nodes {
+                    user {
+                      id 
+                      username 
+                      commitEmail
```

Comment:
How can you guarantee that each user has a not null `commitEmail`. Running the query on my group I don't have a value for all users.

Reply:
This is a fair point, we'll need to dig into this some more - https://docs.gitlab.com/ee/api/graphql/reference/#usercore

Reply:
It looks like users need to select/set their different email attributes on their profile - https://docs.gitlab.com/ee/user/profile/#add-emails-to-your-user-profile

The `/users` REST API only returns `publicEmail` by default (if set) - https://docs.gitlab.com/ee/api/users.html#single-user

The `UserCore` object has a couple of more attributes available, but again, depends on what the user has set on their profile - https://docs.gitlab.com/ee/api/graphql/reference/#usercore

As such, their is no way to guarantee this attribute always has a value. Will reach out to GitLab Support for further  clarification

Reply:
Switched to `publicEmail` as the `email` attribute is deprecated

Reply:
I think we need to fetch the primary email address but there does not seem to be a way to do it easily. I cannot test it cause we are still on self-hosted gitlab and email fiedl of the REST api is currently working fine.